### PR TITLE
feat: [GTM-613]: add new feature banner

### PR DESCRIPTION
**Pull request by nanaxuHarness**
##### Summary:

this is to add feature banner so that each module page can show different messages accordingly, as well as it's behind the FF `FEATURE_ENFORCEMENT_ENABLED`

##### Jira Links:

https://harness.atlassian.net/browse/PLG-613

##### Screenshots:

Screen Shot 2022-01-20 at 5 42 30 PM
Screen Shot 2022-01-20 at 5 41 26 PM
Screen Shot 2022-01-20 at 5 36 44 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by nanaxuHarness** [comment]
retrigger jest

File: src/framework/featureStore/FeaturesFactory.ts [hunk 1]
```diff
@@ -1,11 +1,21 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import type { BannerType } from '@common/layouts/FeatureBanner'
```
Comment (vkbansal) [inline src/framework/featureStore/FeaturesFactory.ts:8]: This is a circular dependency. Please fix it.

File: src/modules/10-common/layouts/FeatureBanner.tsx [hunk 2]
```diff
@@ -9,12 +19,180 @@ import { useFeatures } from '@common/hooks/useFeatures'
...
 
+import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
+import { useStrings } from 'framework/strings'
+import { useFeatureFlag } from '@common/hooks/useFeatureFlag'
+import { FeatureFlag } from '@common/featureFlags'
 import css from './layouts.module.scss'
 
 export const BANNER_KEY = 'feature_banner_dismissed'
 
+export enum BannerType {
```
Comment (vkbansal) [inline src/modules/10-common/layouts/FeatureBanner.tsx:30]: Move this to seperate file to avoid circular dependency

**Review by vkbansal — CHANGES_REQUESTED** [review]

File: src/modules/10-common/layouts/FeatureBanner.tsx (hunk 2 above)
Comment (nanaxuHarness) [reply src/modules/10-common/layouts/FeatureBanner.tsx]: moved to `Constants.ts`

**Review by vkbansal — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**